lib/rubocop/cop/style/redundant_parentheses.rb

```diff
@@ -0,0 +1,63 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for redundant parentheses.
+      #
+      # @example
+      #
+      #   # bad
+      #   (x) if ((y.z).nil?)
+      #
+      #   # good
+      #   x if y.z.nil?
+      #
+      class RedundantParentheses < Cop
+        def on_begin(node)
+          return unless parentheses?(node)
+
+          child_node = node.children.first
+          return offense(node, 'a literal') if literal?(child_node)
+          return offense(node, 'a variable') if variable?(child_node)
+          return offense(node, 'a constant') if constant?(child_node)
+          return unless method_call_with_redundant_parentheses?(child_node)
+
+          offense(node, 'a method call')
+        end
+
+        def offense(node, msg)
+          add_offense(node, :expression, "Don't use parentheses around #{msg}.")
+        end
+
+        def parentheses?(node)
+          node.loc.begin && node.loc.begin.is?('('.freeze)
+        end
+
+        def method_call_with_redundant_parentheses?(node)
+          return false unless method_call?(node)
+          return false if range_end?(node)
+
+          send_node, args = method_node_and_args(node)
+
+          send_node.loc.begin || args.size == 0 || square_brackets?(send_node)
+        end
+
+        def_node_matcher :literal?, '{str sym int float hash array}'
```

Comment:
Don't we already have methods like this somewhere? This is definitely not the first cop that uses them.